doc/development/deprecations.rst

```diff
@@ -51,6 +51,28 @@ Pending deprecations
         some_qfunc(params)
         return qml.expval(Hamiltonian)
 
+* ``Operator.__eq__`` and ``Operator.__hash__`` will now raise a warning to indicate upcoming changes
```

Comment:
I wonder if its worth pointing to any python documentation about the dunder methods.  Easy enough to find plenty of explanations with a simple google search, but it might still be an option.

Reply:
I don't mind leaving this as is personally.